kernel/src/kernel.rs

```diff
@@ -949,7 +950,8 @@ impl Kernel {
                                             }
                                         }
                                     }
-                                    alloc_failure => {
+                                    alloc_failure @ AllocResult::NoAllocation
+                                    | alloc_failure @ AllocResult::SameAllocation => {
```

Comment:
```suggestion
                                    alloc_failure @ (AllocResult::NoAllocation
                                    | AllocResult::SameAllocation) => {
```
nit, this is slightly more concise / better motivates use of an @ binding